specs/metered-guidance-errors.md

```diff
@@ -0,0 +1,151 @@
+- 402s are dependent on the HTTP Verb that is used. In the configuration file, HTTP Verbs need to be explicit.\* 402s should only be returned for the metered endpoints and only for the HTTP Verb used.
+- The list of 402s is available [here](https://learn.microsoft.com/en-us/graph/teams-licenses#payment-related-errors):
+
+| Error code             | Scenario                                         | HTTP Verb    | Sample error message                                                                                      |
+| :--------------------- | :----------------------------------------------- | :----------- | :-------------------------------------------------------------------------------------------------------- |
+| 402 (Payment Required) | Passing `model=A` without a Microsoft E5 license | GET          | `...needs a valid license to access this API...`, `...tenant needs a valid license to access this API...` |
+| 402 (Payment Required) | Calling Patch API passing `model=B`              | PATCH        | `...query parameter 'model' does not support value 'B' for this API. Use billing model 'A'...`            |
+| 402 (Payment Required) | `Evaluation mode` capacity exceeded              | GET or PATCH | `...evaluation mode capacity has been exceeded. Use a valid billing model...`                             |
+
+> **Note**
+> Full error messages are currently not available and we don't have specific information on the ODSP HTTP Errors yet. When we will get them, we will update the plugin to include them.
+
+#### Payment Required
+
+```shell
+[ REQUEST ]    GET https://graph.microsoft.com/v1.0/teams/*/channels/getAllMessages?model=A
+[  CHAOS  ]  ┌ 402 PaymentRequired (Needs a valid license to access this API)
+             └ GET https://graph.microsoft.com/v1.0/teams/*/channels/getAllMessages?model=A
+```
+
+## Configuration file
+
+The plugin will have its own preset file. This configuration file is where metered APIs will be listed. This file should have default values for all supported Microsoft Graph clouds. As of today, only the commercial cloud supports Metered APIs.
+
+- Each `urlToWatch` are rolled up to a single metered API. Multiple `urlToWatch` value means each endpoint behaves the same and should be treated as a metered API.
+- `supportedPaymentModels` are query string passed to the endpoint. If not present, they are considered in Evaluation Mode. If present, they need to be available in the `supportedPaymentModels` list.
+- `billingInformation` represents an URL to get more billing information on the API (if any).
+- `registrationForm` represents an URL to request access to the API (if any).
+
+```json
+{
+  "plugins": [
+    {
+      "name": "GraphMeteredGuidancePlugin",
+      "enabled": true,
+      "pluginPath": "GraphProxyPlugins\\msgraph-developer-proxy-plugins.dll",
+      "configSection": "graphMeteredGuidancePlugin"
+    }
+  ],
+  "graphMeteredGuidancePlugin": [
+    {
+      "urlsToWatch": [

```

Comment:
If `httpVerbs` are not specified do we assume `GET` or all verbs?

Reply:
Added to the config file field definition.